merino-cache/src/memory.rs

```diff
@@ -128,7 +128,18 @@ impl Suggester {
                 timer.tick().await;
                 loop {
                     timer.tick().await;
-                    cloned_suggester.remove_expired_entries();
+                    let mut cache = cloned_suggester.clone();
+
+                    // Dispatch the expiry task to the blocking threads of the
+                    // runtime. This prevents the expiry task, which is inherently
+                    // blocking, from blocking the other tasks running on the
+                    // core threads of the runtime.
+                    //
+                    // Wait for the task to finish so that only one expiry task
+                    // is allowed to run.
+                    let _ = tokio::task::spawn_blocking(move || {
+                        cache.remove_expired_entries();
+                    }).await;
```

Comment:
Looks like await the blocking task (assuming this `JoinHandle` is not blocking) should avoid the repetitive task scheduling concern.